fossunited/foss_profiles/doctype/foss_user_profile/test_foss_user_profile.py

```diff
@@ -36,3 +39,32 @@ def test_add_profile(self):
         profile_exists = frappe.db.exists(USER_PROFILE, {"user": frappe_user.name})
 
         self.assertTrue(profile_exists)
+
+    def test_private_profile_access(self):
+        fake = Faker()
+        test_name = fake.name()
+        test_user = frappe.get_doc(
+            {
+                "doctype": "User",
+                "email": str(uuid.uuid4()) + "@fossunited.org",
+                "first_name": test_name.split(" ")[0],
+                "name": test_name,
+                "full_name": test_name,
+            },
+        ).insert()
+
+        private_profile = frappe.get_doc(
+            {"doctype": USER_PROFILE, "user": test_user.name, "is_private": 1}
+        ).insert()
```

Comment:
```suggestion
        private_profile = frappe.get_doc(
            {
                "doctype": USER_PROFILE,
                "user": test_user.name,
                "is_private": 1,
                "username": fake.user_name(),
            }
        ).insert()
```

username is a mandatory field that we didn't include earlier. the test should pass on CI now. I'll be adding this change to the branch myself.